java/core/src/test/java/sleeper/core/statestore/transactionlog/transactions/TransactionSerDeTest.java

```diff
@@ -0,0 +1,179 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.core.statestore.transactionlog.transactions;
+
+import org.approvaltests.Approvals;
+import org.junit.jupiter.api.Test;
+
+import sleeper.core.partition.PartitionTree;
+import sleeper.core.partition.PartitionsBuilder;
+import sleeper.core.schema.Schema;
+import sleeper.core.schema.type.StringType;
+import sleeper.core.statestore.AllReferencesToAFile;
+import sleeper.core.statestore.FileReference;
+import sleeper.core.statestore.FileReferenceFactory;
+import sleeper.core.statestore.transactionlog.StateStoreTransaction;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toUnmodifiableList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static sleeper.core.schema.SchemaTestHelper.schemaWithKey;
+import static sleeper.core.statestore.AssignJobIdRequest.assignJobOnPartitionToFiles;
+import static sleeper.core.statestore.SplitFileReference.referenceForChildPartition;
+import static sleeper.core.statestore.SplitFileReferenceRequest.splitFileToChildPartitions;
+
+public class TransactionSerDeTest {
+
+    private static void verify(Schema schema, StateStoreTransaction transaction) {
+        // When
+        TransactionSerDe serDe = new TransactionSerDe(schema);
+        TransactionType type = TransactionType.getType(transaction);
+        String json = serDe.toJsonPrettyPrint(transaction);
+
+        // Then
+        assertThat(serDe.toTransaction(type, json)).isEqualTo(transaction);
+        Approvals.verify(json);
+    }
+
+    @Test
+    void shouldSerDeAddFiles() {
+        // Given
+        Schema schema = schemaWithKey("key");
+        PartitionTree partitions = new PartitionsBuilder(schema).singlePartition("root").buildTree();
+        Instant updateTime = Instant.parse("2024-03-26T09:43:01Z");
+        FileReferenceFactory fileFactory = FileReferenceFactory.fromUpdatedAt(partitions, updateTime);
+        StateStoreTransaction transaction = new AddFilesTransaction(
+                AllReferencesToAFile.newFilesWithReferences(Stream.of(
+                        fileFactory.rootFile("file1.parquet", 100),
+                        fileFactory.rootFile("file2.parquet", 200)),
+                        updateTime)
+                        .collect(toUnmodifiableList()),
+                updateTime);
+        verify(schema, transaction);
+    }
```

Comment:
It's a bit odd reading tests without When/Then sections. I know theyve been moved to the verify method, but maybe we could have a `// When / Then` comment before the verify call?

Reply:
Done, thanks